tools/scxtop/src/cli.rs

```diff
@@ -55,6 +55,9 @@ pub struct TuiArgs {
     /// Process to monitor or all.
     #[arg(long, default_value_t = -1)]
     pub process_id: i32,
+    /// Custom perf events colon delimited (ex: "<event_name>:<event and umask ex: 0x023>:<event_type ex: 4>")
```

Comment:
Is there a risk of an observer effect from supporting multiple perf events like this? Do we need a mitigation for this?

Reply:
There's only ever one perf event attached at a time. This just is expanding the available list so that the selectable perf events has a bigger range. On newer processors there's [plenty of counters](https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/programmer-references/58550-0.01.pdf):

> There are six core performance event counters per thread, six performance events counters per L3 complex and sixteen Data Fabric performance events counters mapped to the RDPMC instruction

The thing that will probably need to be added in the future is a differentiation between a CPU and a L3 perf event (since aggregating L3 perf events doesn't really make sense).